ImportBlocks/BlockImporter.cs

```diff
@@ -98,25 +147,49 @@ protected override void OnReceive(object message)
                     _blockchainActorRef = startImport.BlockchainActorRef;
                     _doneAction = startImport.OnComplete;
                     blocksBeingImported = GetBlocksFromFile().GetEnumerator();
+                    stateRootsBeingImported = GetStateRootsFromFile().GetEnumerator();
                     // Start the first import
                     Self.Tell(new Blockchain.ImportCompleted());
                     break;
                 case Blockchain.ImportCompleted _:
                     // Import the next batch
-                    List<Block> blocksToImport = new List<Block>();
-                    for (int i = 0; i < BlocksPerBatch; i++)
+                    if (!(blocksBeingImported is null))
                     {
-                        if (!blocksBeingImported.MoveNext())
+                        List<Block> blocksToImport = new List<Block>();
+                        for (int i = 0; i < BlocksPerBatch; i++)
+                        {
+                            if (blocksBeingImported.MoveNext())
+                            {
+                                blocksToImport.Add(blocksBeingImported.Current);
+                                continue;
+                            }
+                        }
+                        if (blocksToImport.Count > 0)
+                        {
+                            _blockchainActorRef.Tell(new Blockchain.Import { Blocks = blocksToImport });
                             break;
-                        blocksToImport.Add(blocksBeingImported.Current);
+                        }
+                        blocksBeingImported.Dispose();
                     }
-                    if (blocksToImport.Count > 0)
-                        _blockchainActorRef.Tell(new Blockchain.Import { Blocks = blocksToImport });
-                    else
+                    if (!(stateRootsBeingImported is null))
                     {
-                        blocksBeingImported.Dispose();
-                        _doneAction();
+                        List<StateRoot> rootsToImport = new List<StateRoot>();
+                        for (int i = 0; i < RootsPerBatch; i++)
+                        {
+                            if (stateRootsBeingImported.MoveNext())
+                            {
+                                rootsToImport.Add(stateRootsBeingImported.Current);
+                                continue;
+                            }
+                        }
+                        if (rootsToImport.Count > 0)
+                        {
+                            _blockchainActorRef.Tell(new Blockchain.ImportRoots { Roots = rootsToImport });
+                            break;
+                        }
+                        stateRootsBeingImported.Dispose();
```

Comment:
```suggestion
                        stateRootsBeingImported.Dispose();
                        stateRootsBeingImported = null;
```

Please change the block import too